urdf/urdfdom_compatibility.h.in

```diff
@@ -69,16 +69,25 @@ URDF_TYPEDEF_CLASS_POINTER(Sphere);
 URDF_TYPEDEF_CLASS_POINTER(Visual);
 
 URDF_TYPEDEF_CLASS_POINTER(ModelInterface);
+
+typedef boost::shared_ptr<void> VoidSharedPtr;
```

Comment:
I don't see your issue. `VoidSharedPtr `is a datatype declared and used internally in `ColladaParser `only. It's not related to `urdfdom `or `urdf` package. You simply `typedef boost::shared_ptr<void> VoidSharedPtr` just before `collada_parser/src/collada_parser.cpp:179`.